### PR TITLE
PR 1 of n: Molecular Subtyping - HGG (Defining Lesions)

### PR DESCRIPTION
**Pull request by cbethell**
### Purpose/implementation Section

To molecularly subtype HGG samples.

#### What scientific question is your analysis addressing?

What are the samples in the OpenPBTA dataset that fit into the HGG molecular subtypes?

#### What was your approach?

I joined together the data that is relevant to molecular subtyping HGG samples, including the metadata, RNA expression, SNV, and CN data. 

I began by following the plan in the comment [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249#issuecomment-564141215).

#### What GitHub issue does your pull request address?

This PR addresses issue #249.

#### Which areas should receive a particularly close look?

- Does this analysis appear to be correct? 
- Do the variables in the final data.frame seem suffice to molecularly subtype the HGG samples?
- Is there any obvious refactoring needed? 

#### Is there anything that you want to discuss further?

Note: A heatmap displaying the data in this PR is upcoming. 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes, this PR is ready for review. 

#### What types of results are included (e.g., table, figure)?

A `tsv` file with the data in the final data.frame of the R notebook in this PR can be found in the `results` directory of this module at `results/HGG_molecular_subtypes.tsv`.

The table can also be viewed on the html output [here](https://cbethell.github.io/open-pbta-output/01-HGG-molecular-subtyping-data-prep.nb.html).

#### What is your summary of the results?

I have not yet developed a summary of the current results beyond the final data.frame

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### PR Checklist
* [x] Run a linter
* [x] Set the seed (NA)
* [x] Comments and/or documentation up to date 
* [x] Double check your paths
* [x] Spell check any Rmd file or md file
* [x] Restart R and run all notebooks fresh and save

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
@cbethell my read of #249 is that we want a column that tells us about the presence or absence of the following specific mutations:
> _H3F3A_ K28M, _H3F3A_ G35R/V or _HIST1H3B_ K28M

Where the first step is to check _all samples_ for these and then the subsequent steps should include all samples already classified as HGG + any that would be reclassified on the basis of the presence of these mutations.

We also want to check for the presence or absence of
> _IDH1_ R132H 

and looks like _IDH1_ R172, too.

File: analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-data-prep.Rmd [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+    -c(
+      "Kids_First_Participant_ID",
+      "glioma_brain_region",
+      "age_at_diagnosis_days",
+      "germline_sex_estimate"
+    )
+  )  %>%
+  dplyr::group_by(sample_id) %>%
+  dplyr::mutate(
+    OLIG2_expression = paste(sort(unique(OLIG2)), collapse = ", "),
```
[the remark this answers is not in the document]
Comment (cbethell) [reply analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-data-prep.Rmd]: The purpose of this step was to ensure that there was only one row per `sample_id`, however this step was taken out of this particular PR and will be revisited in an upcoming PR (I believe there were duplicate rows for a reason other than the expression values).

**Comment by cbethell** [comment]
Per @jaclyn-taroni's suggestions in [this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/352#pullrequestreview-337189596), this PR is now a notebook that looks only at the HGG defining lesions across all samples.
The upcoming PR will be a script that subsets HGG samples, and a third PR will then incorporate all other relevant data (eg. fusion, CN, RNA expression data).

File: analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+```
+
+## Inconsistencies in disease classification
+
+```{r}
+# Isolate the samples that should be reclassified as HGG
+hgg_samples <- snv_lesions_df %>%
+  dplyr::filter(
+    disease_type_reclassified == "High-grade glioma" &
+      disease_type_new != "High-grade glioma;astrocytoma (WHO grade III/IV)"
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd]: With the v12 data I think this should be
```suggestion
      disease_type_new != "High-grade glioma"
```

**Comment by jaclyn-taroni** [comment]
@cbethell can you provide a link to a rendered version of this updated notebook? I want @jharenza to have a look at the inconsistencies table.

**Comment by cbethell** [comment]
> @cbethell can you provide a link to a rendered version of this updated notebook? I want @jharenza to have a look at the inconsistencies table.

The rendered version of this updated notebook can be found [here](https://cbethell.github.io/open-pbta-output/01-HGG-molecular-subtyping-defining-lesions.nb.html).

File: analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  dplyr::distinct() %>%
+  dplyr::mutate(
+    disease_type_reclassified = dplyr::case_when(
+      H3F3A.K28M == "Yes" |
+        HIST1H3B.K28M == "Yes" |
+        H3F3A.G35R == "Yes" |
+        H3F3A.G35V == "Yes" ~ "High-grade glioma",
+      TRUE ~ as.character(disease_type_new)
+    )
+  )
```
Comment (jharenza) [inline analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd]: Can you make the reclassifications further subdivided - something like: 
`HGG, H3 K28 mutant` or `High-grade glioma, H3 K28 mutant`
`HGG, H3 G35 mutant` or `High-grade glioma, H3 G35 mutant`

Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd]: I've made that update in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/352/commits/a10cdeda7559232acb990392bfe34399c182095f. Note that this table is not the final table from this module, but an interim product https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/352#pullrequestreview-337189596.

**Comment by jaclyn-taroni** [comment]
I'm going to dismiss @jharenza's review because those comments have now been addressed (https://github.com/AlexsLemonade/OpenPBTA-analysis/commit/a10cdeda7559232acb990392bfe34399c182095f) and I would like to reduce the backlog of pull requests we have from before the break.

**Pull request merged**